### ISSUE TITLE
Closed door often show as "opening"

### ISSUE THREAD

**Issue by nklint** [KraigM/homebridge-gpio-garagedoor#7]
When I open up the Home app my garage is often showing the status "opening" or "closing" even though it is actually closed. I know that the sensor is ideed giving the correct state.
Clicking the button corrects the state to showing "closed" as it should be.

**Comment by nklint** [comment]
I'm thinking that this might be related to the fact that I often control the garage port direct with the built in controls.
I have seen that this seems to confuse the state in HomeKit.
But the bottom line here is that the actual state "closed", from the sensor should always override the made up states "opening" and closing witch have to be temporary..

**Comment by stevenguh** [comment]
TL;DR. Try updating your homebridge.

I don't know what version you got, but I just updated the my raspberry pi today with all the latest including homebridge and this plugin.
Most of the annoying bugs I had in the past is either gone, or improved after I updated. For example, using Siri to close garage will timeout perviously. This bug is gone after I update today.

**Comment by nklint** [comment]
This seemes to be resolved by updating to version 0.2.2. Thanks!

**Comment by simonrb2000** [comment]
How can you tell what version homebridge your running? I updated to latest today I think and I'm still getting this.. (Saying garage door is closing when its closed) It used to work but recently it now says closing..

**Comment by stevenguh** [comment]
I am running hombridge 0.4.9 and homebridge-gpio-garagedoor 0.2.2.
You can use `npm list -g | grep homebridge` to see your version.

**Comment by simonrb2000** [comment]
Thanks mine is 0.4.9, garage door is 1.0.2.. the door shows as closed but if I ask Siri she says yourngarage door is closing.. she used to say closed and it does show as closed - even in Eve. Just Siri says closing..

**Comment by simonrb2000** [comment]
Can't see 1.0.2 as a version ever released.. 

pi@raspberrypi ~ $ npm list -g | grep homebridge
├─┬ homebridge@0.4.9
├─┬ homebridge-legacy-plugins@0.1.0
├─┬ homebridge-nest@1.1.2
├── homebridge-rasppi-gpio-garagedoor@1.0.2

**Comment by stevenguh** [comment]
I think you installed [homebridge-rasppi-gpio-garagedoor](https://github.com/benlamonica/homebridge-rasppi-gpio-garagedoor), which is not the same plug-in as in this repo.

**Comment by simonrb2000** [comment]
Your quite right! my bad, how embarrassing lol. Is it the same thing? Wondered if its worth changing to this plugin lol

**Comment by stevenguh** [comment]
lol. As for functionality wise, they might be similar. But I wouldn't not say they are the same. This package use a npm package called onoff to control gpio instead of using bash script. The installation on this package is a lot easier since it doesn't use the bash script. 

Since you installed homebridge-rasppi-gpio-garagedoor, you might need to revert the installation step you did if you decided to switch to this package.

**Comment by simonrb2000** [comment]
Ah ok. Very informative thank you :-)

As it uses on/off is there a chance if the raspberry Pi is restarted it may open my garage door? Currently this doesn't happen thankfully :-)

Sent from my iPhone

On 22 Nov 2016, at 20:29, Steven Guh <notifications@github.com<mailto:notifications@github.com>> wrote:

lol. As for functionality wise, they might be similar. But I wouldn't not say they are the same. This package use a npm package called onoff to control gpio instead of using bash script. The installation on this package is a lot easier since it doesn't use the bash script.

Since you installed homebridge-rasppi-gpio-garagedoor, you might need to revert the installation step you did if you decided to switch to this package.

##

**Comment by stevenguh** [comment]
In my experience, that has never happened to me.
